dataclients/kubernetes/ingressdefinitions.go

```diff
@@ -83,7 +85,11 @@ type endpointList struct {
 }
 
 func formatEndpoint(a *address, p *port, protocol string) string {
-	return fmt.Sprintf("%s://%s:%d", protocol, a.IP, p.Port)
+	ip := net.ParseIP(a.IP)
+	if strings.Count(a.IP, ":") >= 2 { // our naive IPv6 check
+		return fmt.Sprintf("%s://[%s]:%d", protocol, ip.String(), p.Port)
+	}
+	return fmt.Sprintf("%s://%s:%d", protocol, ip.String(), p.Port)
```

Comment:
There is https://pkg.go.dev/net#JoinHostPort so maybe it should be something like
```go
return protocol + "://" + net.JoinHostPort(a.IP, strconv.Itoa(p.Port))
```

Reply:
done